### PR TITLE
#1362 implement dropdown for proposals

### PR DESCRIPTION
**Pull request by SebinSong**
working on the issue #1362

### CODE REVIEW

File: frontend/views/containers/dashboard/GroupMembersAllModal.vue [hunk 1]
```diff
@@ -94,8 +94,14 @@ modal-base-template.has-background(ref='modal' :fullscreen='true' :a11yTitle='L(
                 i18n.pill.is-primary(v-else-if='isNew' data-test='pillNew') new
 
             .c-actions
+              button.is-icon(
+                v-if='$route.query.toRemove'
+                @click.stop='removeMember'
```
Comment (taoeffect) [inline frontend/views/containers/dashboard/GroupMembersAllModal.vue]: So this `removeMember` method doesn't actually do anything. Do you think you can fill it in?

File: frontend/views/containers/chatroom/GroupMembersDirectMessages.vue [hunk 2]
```diff
@@ -66,7 +66,7 @@ import Search from '@components/Search.vue'
 import AvatarUser from '@components/AvatarUser.vue'
 
 export default ({
-  name: 'GroupMembersAllModal',
+  name: 'GroupMembersDirectMessages',
```
Comment (taoeffect) [inline frontend/views/containers/chatroom/GroupMembersDirectMessages.vue:69]: Nice catch! 👍

File: frontend/views/components/ButtonDropdownMenu.vue [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      type: Array
+      /**
+       * NOTE: Shape of the array must strictly follow below statement.
+       *
+       * - An item to be mapped to a 'menu-item' component:
+       *  { type: 'item', id: string, name: string, icon: string }
+       *
+       * - An item to be mapped to a 'menu-header' component:
+       *  { type: 'header', name: string }
+       */
```
Comment (taoeffect) [inline frontend/views/components/ButtonDropdownMenu.vue]: Nice, clear comment 👍

**Review by taoeffect — CHANGES_REQUESTED** [review]
Thanks for sending this in @SebinSong!

I couldn't figure out how to test this though, as the proposals widget doesn't show up at all in one case, and when there's a proposal the dropdown menu still doesn't seem to be visible either:

Screen Shot 2022-08-03 at 3 26 56 PM

Screen Shot 2022-08-03 at 3 28 53 PM

**Comment by SebinSong** [comment]
@taoeffect 
I have no idea how you got that 'Create Generic proposal' button. are you testing it in the correct branch?
[`sebin/task/#1362-implement-dropdown-for-proposals`](https://github.com/okTurtles/group-income/tree/sebin/task/%231362-implement-dropdown-for-proposals)

Below is what I see in Dashboard(I replaced that button with this dropdown). when there is a proposal opened.
image

**Review by taoeffect — CHANGES_REQUESTED** [review]
Sebin - you're right, I must have been testing some other PR by accident, thinking it was this one. Doh! 😖

OK, so - going back to this PR. Then it seems like there's two main issues:

1. The `GroupMembersAllModal.vue` should implement `removeMember`. It should be easy, just copy/paste the code that exists in the button for "Remove Member" when you click on a username.
2. As shown in your bottom screenshot - for some reason there's additional text beneath the proposals starting with "In Group Income..." That needs to not be there.

**Comment by SebinSong** [comment]
@taoeffect Thanks,


> The GroupMembersAllModal.vue should implement removeMember. 

Done.

> As shown in your bottom screenshot - for some reason there's additional text beneath the proposals starting with "In Group Income..." That needs to not be there.

Done.

And also worked on resolving below issue. 

> I couldn't figure out how to test this though, as the proposals widget doesn't show up at all in one case

The dashboard does show this `CalloutCard` now when there's no proposals to display.

image

 _But this leads to the following question._

**Q.** Don't we need to display some of the items under certain conditions? Like what I wrote in below screenshot?(Pls go ahead correct me if any of them are wrong)
image

Cheers,

**Comment by taoeffect** [comment]
@SebinSong - great updates!

And yes - I agree with your proposal above, especially because with it it, it seems weird things happen like this:

Screenshot 2022-08-04 at 16-50-20 Group Dashboard

**Comment by SebinSong** [comment]
@taoeffect updated the PR with a change where some menu items are disabled under certain conditions.

**Review by taoeffect — APPROVED** [review]
Excellent work @SebinSong! Approved!

**Pull request merged**